src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -653,79 +654,78 @@ void ov::npuw::CompiledModel::CompiledModelDesc::deserialize(std::istream& strea
     LOG_DEBUG("DONE.");
 }
 
-void ov::npuw::CompiledModel::serialize(std::ostream& stream) const {
-    LOG_INFO("Serializing CompiledModel...");
-    LOG_BLOCK();
-
+void ov::npuw::CompiledModel::export_model(std::ostream& stream) const {
     using namespace ov::npuw::s11n;
 
-    // Serialize name
-    write(stream, m_name);
-
-    // Serialize inputs and outputs
-    write(stream, inputs());
-    write(stream, outputs());
-
-    // Serialize meta
-    write(stream, m_inputs_to_submodels_inputs);
-    write(stream, m_outputs_to_submodels_outputs);
-    write(stream, m_param_subscribers);
-    write(stream, m_submodels_input_to_prev_output);
-
-    // Write device list
-    write(stream, m_dev_list);
-
-    // Write config
-    write(stream, m_cfg);
-    // FIXME: utilize overload instead
-    write(stream, m_non_npuw_props.size());
-    for (const auto& p : m_non_npuw_props) {
-        // Skip properties which don't need to/can't be serialized
-        // FIXME: extend the logic
-        if (p.first == ov::cache_encryption_callbacks.name()) {
-            write(stream, false);
-            continue;
-        }
-        write(stream, true);
-        write(stream, p.first);
-        write_any(stream, p.second);
+    // Identify encryption flow
+    bool encryption_required = false;
+    EncryptionCallbacks enc_callbacks;
+    if (auto it = m_non_npuw_props.find(ov::cache_encryption_callbacks.name());
+        it != m_non_npuw_props.end() && it->second.as<EncryptionCallbacks>().encrypt) {
+        LOG_INFO("Encryption will be done via the function provided.");
+        encryption_required = true;
+        enc_callbacks.encrypt = it->second.as<EncryptionCallbacks>().encrypt;
     }
 
-    // Write flow identifier
+    // Identify either full flow or weightless
     bool is_weightless = true;
-    if (m_non_npuw_props.count(ov::cache_mode.name()) &&
-        m_non_npuw_props.at(ov::cache_mode.name()).as<CacheMode>() == CacheMode::OPTIMIZE_SPEED) {
+    if (auto it = m_non_npuw_props.find(ov::cache_mode.name());
+        it != m_non_npuw_props.end() && it->second.as<CacheMode>() == CacheMode::OPTIMIZE_SPEED) {
+        LOG_INFO("Serialization will be done via flow with weights.");
         is_weightless = false;
     }
+
+    // Write header regardless of encryption requirement - to identify NPUW serializated blobs
+    // Serialize magic number first
+    write(stream, NPUW_SERIALIZATION_INDICATOR);
+    // Serilize CompiledModel identifier
+    write(stream, NPUW_COMPILED_MODEL_INDICATOR);
```

Comment:
It breaks the compatibility with the previous blobs I believe?

Reply:
Yes